src/main/java/ch/njol/skript/test/runner/TestMode.java

```diff
@@ -67,4 +67,9 @@ public class TestMode {
 	 */
 	public static boolean docsFailed;
 
+	/**
+	 * True when test elements should be registered, false if not.
+	 */
+	public static final boolean TEST_ELEMENTS = ENABLED && !GEN_DOCS;
```

Comment:
instead of using a field, could we check for the `NoDoc` annotation on the class?

Reply:
So instead of these checks you mean doing `ExprX.class.getAnnotation(NoDoc.class) != null`?

Reply:
yeah, in the jsongenerator. like we do here: https://github.com/SkriptLang/Skript/blob/38d318726a2c61ec09b5386c79de58ac82596bdd/src/main/java/ch/njol/skript/doc/JSONGenerator.java

Reply:
should be good now :+1: